cpp/src/sort/is_sorted.cu

```diff
@@ -44,16 +46,22 @@ auto is_sorted(cudf::table_view const& in,
   if (cudf::detail::has_nested_columns(in)) {
     auto const device_comparator = comparator.less<true>(has_nested_nulls(in));
 
-    return thrust::is_sorted(rmm::exec_policy(stream),
-                             thrust::make_counting_iterator(0),
-                             thrust::make_counting_iterator(in.num_rows()),
-                             device_comparator);
+    auto d_results = rmm::device_uvector<bool>(in.num_rows(), stream);
+    thrust::transform(rmm::exec_policy(stream),
```

Comment:
```suggestion
    // Use transform and count instead of thrust's is_sorted to improve compile time.
    thrust::transform(rmm::exec_policy(stream),
```